### PR TITLE
add SQLServerDB support for ReadFile

### PR DESCRIPTION
**Pull request by hz-lschick**
This PR adds SQL Server support for command ReadFile.

This is done by using the bcp command, officially supported from Microsoft for Linux and Mac via the mssql-tools.

### CODE REVIEW

File: mara_pipelines/commands/files.py [hunk 1]
```diff
@@ -57,21 +57,37 @@ def db_alias(self):
...
-        if not isinstance(mara_db.dbs.db(self.db_alias()), mara_db.dbs.BigQueryDB):
-            return \
-                f'{uncompressor(self.compression)} "{pathlib.Path(config.data_dir()) / self.file_name}" \\\n' \
-                + (f'  | {shlex.quote(sys.executable)} "{self.mapper_file_path()}" \\\n'
-                   if self.mapper_script_file_name else '') \
-                + ('  | sort -u \\\n' if self.make_unique else '') \
-                + '  | ' + copy_from_stdin_command
+        file_path = pathlib.Path(config.data_dir()) / self.file_name
+        db = mara_db.dbs.db(self.db_alias())
+        if isinstance(db, mara_db.dbs.SQLServerDB):
```
Comment (jankatins) [inline mara_pipelines/commands/files.py:62]: That this generic command needs multiple branches for specific DBs is IMO a sign that the abstraction here is not yet right. If someone has a DB which is more similar to BQ/SQLServer, you are out of luck. Any chance to clean this up? E.g. move the specifics to a singledispatch functions (if BQ and SQL Server actually have some similarity) and either branch on some flag or simply call first one singledispatch function and if that raises NotImplmentedError try another?

Comment (hz-lschick) [reply mara_pipelines/commands/files.py:62]: I agree. I did some researches. `functools` doesn't have a `singledispatch` for methods. I found serveral solutions [at stack overflow](https://stackoverflow.com/questions/24601722/how-can-i-use-functools-singledispatch-with-instance-methods).

What is your opinion, should we use an on implementation (first comment) or the module [`multipledispatch`](https://pypi.org/project/multipledispatch/)? @jankatins 


I think about adding a new method `read_file_command(self, db: object)` which is then called like this:
``` python
def shell_method(self):
    return self.read_file_command(mara_db.dbs.db(self.db_alias()))
```
`read_file_command` would then have a default implementation using `copy_from_stdin_command(...)` and have custom implementations for BigQueryDB and SQLServerDB

Comment (jankatins) [reply mara_pipelines/commands/files.py:62]: I would just move the things to a function and pass whatever is needed as context to the function `whatever(source DB, ...)` and use `@singledispatch` or multidispatch if source + target are stuff to dispatch on (hopefully not). E.g. 

```
@singledispatch
def read_file_command(db_alias:str):
    return read_file_command(mara_db.dbs.db(db_alias))

@read_file_command.register(DB)
def read_file_command(db_alias:str):
    return ... current default

@read_file_command.register(BigQueryDB)
def read_file_command(db_alias:str):
    return ... BQ specific ...
````

?

**Comment by hz-lschick** [comment]
@jankatins I have now a better solution and will close this PR. It is possible to get the piped data from `/dev/stdin`, so, there is no need for this custom implementation. I added a new PR here: https://github.com/mara/mara-db/pull/41

`bcp` is part of the [SQL Server command line tools available for Linux](https://docs.microsoft.com/en-us/sql/tools/overview-sql-tools?view=sql-server-ver15#command-line-tools), so, this is the first step moving into the direction of https://github.com/mara/mara-db/issues/39

I have to acknowledge that I never linked the implementation for BigQuery either, but @gathineou needed it. I asked him to check if he can use `/dev/stdin` as well (#50).

**Pull request closed without merging**